src/signals/incident-management/components/FilterForm/index.js

```diff
@@ -82,19 +82,22 @@ const FilterForm = ({
     onClearFilter();
   }, [onClearFilter]);
 
-  const onChangeForm = useCallback(() => {
-    if (isNewFilter) {
-      return;
-    }
+  const onChangeForm = useCallback(
+    event => {
+      if (isNewFilter) {
+        return;
+      }
 
-    const formData = parseOutputFormData(formRef.current);
-    const valuesHaveChanged = !isEqual(formData, filterData);
-    const btnHasSaveLabel = submitBtnLabel === saveSubmitBtnLabel;
+      const formData = parseOutputFormData(event.form);
+      const valuesHaveChanged = !isEqual(formData, filterData);
+      const btnHasSaveLabel = submitBtnLabel === saveSubmitBtnLabel;
 
-    if (valuesHaveChanged && !btnHasSaveLabel) {
-      setSubmitBtnLabel(saveSubmitBtnLabel);
-    }
-  }, [filterData, isNewFilter, submitBtnLabel]);
+      if (valuesHaveChanged && !btnHasSaveLabel) {
```

Comment:
Is this necessary in the callback? Can't this be checked in the render step? Looks like changing the button label will cause an extra rerender now.

Reply:
No, has to be in the `onChangeForm` callback; the value of the form fields has to be compared to those that are in the state after something has changed in the form. That't not possible by moving it to the render step.